src/wh_flash_ramsim.c

```diff
@@ -0,0 +1,203 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+
+#include "wolfhsm/wh_error.h"
+#include "wolfhsm/wh_flash_ramsim.h"
+
+
+static bool isMemoryErased(whFlashRamsimCtx* context, uint32_t offset,
+                           uint32_t size);
+
+
+static bool isMemoryErased(whFlashRamsimCtx* context, uint32_t offset,
+                           uint32_t size)
+{
+    for (uint32_t i = 0; i < size; ++i) {
+        if (context->memory[offset + i] != context->erasedByte) {
+            return false;
+        }
+    }
+    return true;
+}
+
+
+/* Simulator functions */
+int whFlashRamsim_Init(void* context, const void* config)
+{
+    whFlashRamsimCtx*       ctx = (whFlashRamsimCtx*)context;
+    const whFlashRamsimCfg* cfg = (const whFlashRamsimCfg*)config;
+
+    if (context == NULL || config == NULL) {
+        return WH_ERROR_BADARGS;
+    }
+
+    ctx->size       = cfg->size;
+    ctx->sectorSize = cfg->sectorSize;
+    ctx->pageSize   = cfg->pageSize;
+    ctx->memory     = (uint8_t*)malloc(ctx->size);
+    ctx->erasedByte = cfg->erasedByte;
+
+    if (!ctx->memory) {
+        return WH_ERROR_BADARGS;
+    }
+
+    /* Simulate starting from erased flash */
+    memset(ctx->memory, ctx->erasedByte, ctx->size);
+
+    return WH_FLASH_RAMSIM_OK;
+}
+
+int whFlashRamsim_Cleanup(void* context)
+{
+    whFlashRamsimCtx* ctx = (whFlashRamsimCtx*)context;
+
+    if (context == NULL) {
+        return WH_ERROR_BADARGS;
+    }
+
+    free(ctx->memory);
```

Comment:
```suggestion
    if (ctx->memory != NULL) {
        free(ctx->memory);
        ctx->memory = NULL;
    }
```
stdlib free() can't handle a NULL argument.

Reply:
fixed